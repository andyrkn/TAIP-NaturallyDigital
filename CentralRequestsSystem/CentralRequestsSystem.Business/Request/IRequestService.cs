using CentralRequestsSystem.Business.Request.Models;
using System.Threading.Tasks;

namespace CentralRequestsSystem.Business.Request
{
    public interface IRequestService
    {
        Task AddRequest(AddRequestModel addRequestModel);
    }
}
