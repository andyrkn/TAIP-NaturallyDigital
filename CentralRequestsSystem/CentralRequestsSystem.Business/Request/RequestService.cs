using System;
using System.Threading.Tasks;
using CentralRequestsSystem.Business.Request.Models;
using CentralRequestsSystem.Core;

namespace CentralRequestsSystem.Business.Request
{
    public class RequestService : IRequestService
    {
        private readonly IRequestRepository _requestRepository;
        public Task AddRequest(AddRequestModel addRequestModel)
            => throw new NotImplementedException();
    }
}
