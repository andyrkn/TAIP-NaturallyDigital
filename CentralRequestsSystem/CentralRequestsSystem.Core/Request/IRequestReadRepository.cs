using CentralRequestsSystem.Core.Repository;

namespace CentralRequestsSystem.Core
{
    public interface IRequestReadRepository : IReadRepository<Request>
    {
    }
}
