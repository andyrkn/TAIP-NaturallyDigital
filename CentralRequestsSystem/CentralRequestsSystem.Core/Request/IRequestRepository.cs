using CentralRequestsSystem.Core.Repository;

namespace CentralRequestsSystem.Core
{
    public interface IRequestRepository : IWriteRepository<Request>
    {
    }
}
