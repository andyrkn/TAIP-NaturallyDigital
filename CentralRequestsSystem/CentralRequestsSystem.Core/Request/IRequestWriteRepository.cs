using CentralRequestsSystem.Core.Repository;

namespace CentralRequestsSystem.Core
{
    public interface IRequestWriteRepository : IWriteRepository<Request>
    {
    }
}
