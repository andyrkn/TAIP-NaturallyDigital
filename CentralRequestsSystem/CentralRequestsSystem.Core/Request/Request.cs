using CSharpFunctionalExtensions;
using System;

namespace CentralRequestsSystem.Core
{
    public sealed class Request : Entity
    {
        private Request()
        {

        }

        public string UserAdress { get; private set; }
        public string IdentityProviderAdress { get; private set; }
        public DateTime Date { get; private set; }
        public string Payload { get; private set; }

        public static Result<Request> CreateRequest(string userAdress, string identityProviderAdress, DateTime date, string payload)
            => Result.Ok(new Request
                {
                    UserAdress = userAdress,
                    IdentityProviderAdress = identityProviderAdress,
                    Date = date,
                    Payload = payload
                });
    }
}
