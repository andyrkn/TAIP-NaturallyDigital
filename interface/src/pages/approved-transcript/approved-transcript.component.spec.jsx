import React from 'react';
import { assert } from "chai";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import ApprovedTranscript from './approved-transcript.component';
import Request from "../../components/request/request.component";
import Response from "../../components/response/response.component";

configure({
    adapter: new Adapter()
});

describe("Approved transcript page", () => {

    let getAccountAddress, getRequest, match, uploadContent, createIdentity, deleteRequest;

    beforeEach(() => {
        getAccountAddress = jest.fn(() => new Promise(() => "account"));
        getRequest = jest.fn((id) => new Promise(() => { }));
        createIdentity = jest.fn((address, id, ip) => new Promise(() => "hash"));
        uploadContent = jest.fn((content) => new Promise(() => "hash"));
        deleteRequest = jest.fn((id) => new Promise(() => { }));
        match = { params: { id: 1 } };
    })

    it("makes a snapshot", () => {
        const tree = renderer.create(<ApprovedTranscript />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders the page components correctly", () => {
        const wrapper = shallow(<ApprovedTranscript match={match} getRequest={getRequest} getAccountAddress={getAccountAddress} />)
        const request = wrapper.find(Request);
        const response = wrapper.find(Response);
        const confirmButton = wrapper.find("button");

        assert.lengthOf(request, 1);
        assert.lengthOf(response, 1);
        assert.lengthOf(confirmButton, 1);
    });

    it("calls upload on IPFS, create identity method on contract and delete request when confirm button is clicked", () => {
        let wrapper = shallow(<ApprovedTranscript  match={match} getRequest={getRequest} getAccountAddress={getAccountAddress} 
            uploadContent={uploadContent} createIdentity={createIdentity} deleteRequest={deleteRequest}/>)

        wrapper.find("button").at(0).simulate("click");

        expect(uploadContent.mock.calls.length).toBe(1);
        expect(createIdentity.mock.calls.length).toBe(0);
        expect(deleteRequest.mock.calls.length).toBe(0);
    })
})