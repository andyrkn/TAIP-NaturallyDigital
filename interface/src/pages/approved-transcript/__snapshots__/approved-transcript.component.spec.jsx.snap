// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Approved transcript page makes a snapshot 1`] = `
Array [
  <div
    className="makeStyles-root-1"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-nav-2 MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        className="MuiBox-root MuiBox-root-51 makeStyles-box-6"
      >
        <img
          alt="DSC-Logo"
          className="navbar-logo"
          src="spiderman.png"
        />
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root makeStyles-icon-7"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
      </div>
    </header>
  </div>,
  <main
    className="main"
  >
    <h2>
      Request
    </h2>
    <div>
      <span
        className="key"
      >
        Institution: 
      </span>
      <span>
        
      </span>
    </div>
    <div>
      <span
        className="key"
      >
        Request type: 
      </span>
      <span>
        
      </span>
    </div>
    <h3>
      Response
    </h3>
    <div>
      <button
        className="button button2"
        onClick={[Function]}
        type="button"
      >
        Accept response and save
      </button>
    </div>
    <div
      className="error"
    >
      
    </div>
  </main>,
]
`;
